prime-dotnet-webapi/ApiDbContext.cs

```diff
@@ -204,11 +204,27 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 .WithMany(e => e.AdjudicatorNotes)
                 .HasForeignKey(an => an.EnrolleeId);
 
-            modelBuilder.Entity<AccessTerm>()
+            modelBuilder.Entity<Agreement>()
                 .HasOne(toa => toa.Enrollee)
-                .WithMany(e => e.AccessTerms)
+                .WithMany(e => e.Agreements)
                 .HasForeignKey(toa => toa.EnrolleeId);
 
+            modelBuilder.Entity<Agreement>()
+                .HasOne(toa => toa.Organization)
+                .WithMany(e => e.Agreements)
+                .HasForeignKey(toa => toa.OrganizationId);
+
+            modelBuilder.Entity<Agreement>()
+                .HasOne(toa => toa.Party)
+                .WithMany(e => e.Agreements)
+                .HasForeignKey(toa => toa.PartyId);
+
+            modelBuilder.Entity<Agreement>()
+                .HasCheckConstraint("CHK_OnlyOneForeignKey",
+                    "( CASE WHEN \"EnrolleeId\" IS NULL THEN 0 ELSE 1 END"
```

Comment:
You could use a string literal here to make this look slightly nicer. 
`@"(CASE WHEN ""EnrolleeId"" IS NULL THEN 0 ELSE 1 END
   + CASE WHEN ""OrganizationId"" IS NULL THEN 0 ELSE 1 END
   + CASE WHEN ""PartyId"" IS NULL THEN 0 ELSE 1 END) = 1"`